erc7824-docs/docs/guides/migration-guide.md

```diff
@@ -12,6 +12,419 @@ import TabItem from '@theme/TabItem';
 
 If you are coming from an earlier version of Nitrolite, you will need to account for the following breaking changes.
 
+## 0.5.x Breaking changes
+
+The 0.5.x release includes fundamental protocol changes affecting session keys, channel operations, state signatures, and channel resize rules. The main objective of these changes is to enhance security, and provide better experience for developers and users by ability to limit allowances for specific applications.
+
+**Not ready to migrate?** Unfortunately, at this time Yellow Network does not provide ClearNodes running the previous version of the protocol, so you will need to migrate to the latest version to continue using the Network.
+
+### Protocol Changes
+
+These protocol-level changes affect all implementations and integrations with the Yellow Network.
+
+#### Session Keys: Applications, Allowances, and Expiration
+
+Session keys now have enhanced properties that define their access levels and capabilities:
+
+- **Application field**: Determines the scope of session key permissions. Setting this to an application name (e.g., "My Trading App") grants application-scoped access with enforced allowances. Setting it to "clearnode" grants root access equivalent to the wallet itself.
+
+- **Allowances field**: Defines spending limits for application-scoped session keys. These limits are tracked cumulatively across all operations and are enforced by the protocol.
+
+- **Expires_at field**: Uses a bigint timestamp (seconds since epoch). Once expired, session keys are permanently frozen and cannot be reactivated. This is particularly critical for root access keys (application set to "clearnode") - if they expire, you lose the ability to perform channel operations.
+
+#### Channel Creation: Separate Create and Fund Steps
+
+Clearnode no longer supports creating channels with an initial deposit. All channels must be created with zero balance and funded separately through a resize operation. This two-step process ensures cleaner state management and prevents edge cases in channel initialization.
+
+#### State Signatures: Wallet vs Session Key Signing
+
+A fundamental change in how channel states are signed:
+
+- **Channels created before v0.5.0**: The participant address is the session key, and all states must be signed by that session key.
+
+- **Channels created after v0.5.0**: The participant address is the wallet address, and all states must be signed by the wallet.
+
+This change improves security and aligns with standard practices, but requires careful handling during the transition period.
```

Comment:
Let's highlight more that this only affects the on-chain.

```suggestion
A fundamental change in how on-chain channel states are signed:

- **Channels created before v0.5.0**: The on-chain participant address is set to a session key, and all states must be signed by that session key.

- **Channels created after v0.5.0**: The on-chain participant address is set to a wallet address, and all states must be signed by the wallet.

This change improves security and aligns with standard practices, but requires careful handling during the transition period.

Please note that session keys logic is still related in the same way to the off-chain operations (Clearnode and application sessions).
```

Reply:
Or does it sound more confusing with the "on-chain" phrase?

Reply:
Yeah, when I read "on-chain channel", it sounds to me that there are somewhere off-chain channels)